First attempts to update an existing alias. I've created the whole str…

litosnano opened the pull request:
…ucture but with two issues that I need some advice.

Commit:
First attemps to update an existing alias. I've created the whole structure but with two issues that I need some advice.

adriacidre commented on config/config.go:
```diff
@@ -35,6 +35,16 @@ func (c *Config) Get(name string) (*Command , error){
 	return nil, errors.New("")
 }
 
+func (c *Config) Position(name string) (int){
```
Given now we have unique names for each command, is better if you manage all the stuff related to the config file inside the config.go file, so, a method like
```
func (c *Config) UpdateCommand(name string, cmd *Command) error {
    ....
}
```
will be easy to maintain

adriacidre commented on config/config.go:
```diff
@@ -35,6 +35,16 @@ func (c *Config) Get(name string) (*Command , error){
+			return pos
+		}
+	}
+	return -1
```
As go can return multiple arguments it kind of a standard to return 2 arguments when there is something like this. So:
```
func (c *Config) Position(name string) (int, error){
   if (name == "supu") {
     return 1, nil
   }
   return 0, errors.New("it does not exist")
}
```

adriacidre commented on main.go:
```diff
@@ -146,6 +148,40 @@ func rm(c config.Config, path, name string) {
+		return
+	}
+
+	search := c.Position(name)
```
Instead of `c.Postion(name)` and the loop below you can simply use `cmd, err := c.Get(name)`, so you reuse all the logic on the `Config`.

Once you have the command (cmd) is time to convert it on a valid string (the way is stored on the file). This logic is already implemented on `Config.toLines()`. However is not accessible, so we will need to do a refactor there. Ideally I would like to see something like this:

```go
// Get the command you want to update
currentAlias, err := c.Get(name)
if err != nil {...}

// Dispay the editor with the command on it to capture the input
newAlias, err := editor.CaptureInputFromEditor(
    editor.GetPreferredEditorFromEnvironment,
    currentAlias.String(), // This is where you have to move the toLines() logic
)
if err != nil {...}

// Replace existing command
err = c.Update(name, newAlias)
if err != nil {...}

// Save the configuration to a file
c.Save()
```

We can do some pair programming if you like :)

adriacidre commented on main.go:
```diff
@@ -146,6 +148,40 @@ func rm(c config.Config, path, name string) {
+		c.HeaderTemplate(name),
+	)
+	if err != nil {
+		panic(err)
```
avoid (in general) working with panicss

adriacidre replied:
this is still a problem :P

Commit:
Added an update for an existing alias.
New update command done having an existing name.
New functions (ToLines and Byte) has been created

adriacidre commented on config/config.go:
```diff
@@ -180,6 +199,17 @@ func (c *Config) RmCommand(name string) error {
+			return nil
+		}
+	}
+	return errors.New("")
```
Can we return an error string here?

adriacidre commented on config/config.go:
```diff
@@ -91,29 +91,48 @@ func (c *Config) parseLines(lines []string) {
+	return lines
+}
+
+// Byte takes the current lines from ToLines
```
I would like to have *Config type* and its related methods on `config.go` while *Command* and its related methods on `command.go` file. Do you think you can split them?

adriacidre commented on config/config.go:
```diff
@@ -91,29 +91,48 @@ func (c *Config) parseLines(lines []string) {
+	return tbytes
+}
+
+func (c *Command) ToLines() []string {
```
Let's add documentation to all new methods :)

adriacidre reviewed (commented):
Looks nice, let's see if we can clean it up with documentation.
I would like to also see some tests for the new methods on `Command`, let me know if you need a hand with this :)
